src/main/java/com/exasol/containers/ExasolDockerImageReference.java

```diff
@@ -12,22 +13,14 @@
 public class ExasolDockerImageReference {
     private static final Pattern DOCKER_IMAGE_VERSION_PATTERN = Pattern
             .compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:-d(\\d+))?");
+    private static final Pattern MAJOR_VERSION_PATTERN = Pattern
+            .compile(Pattern.quote(EXASOL_DOCKER_IMAGE_ID) + ":(\\d+).*");
```

Comment:
Here, I think you can use the previous DOCKER_IMAGE_VERSION_PATTERN pattern. From looking into the pattern, you should be able to get the major version. In the getMajorVersion method it gets the first matched group from the version.

Could you please check it, if you have not already?

Reply:
I just checked it again. the difference is that `MAJOR_VERSION_PATTERN` only matches on the official docker db. For that I added the `Pattern.quote(EXASOL_DOCKER_IMAGE_ID)` which renders to `exasol/docker-db`.

Reply:
Okay, makes sense. So, not to match on custom images.

Reply:
:+1: